app/src/pages/inside/historyPage/historyView/historyTable/historyTable.jsx

```diff
@@ -78,98 +70,75 @@ export class HistoryTable extends Component {
   static propTypes = {
     intl: intlShape.isRequired,
     historyDepth: PropTypes.string.isRequired,
-    items: PropTypes.array,
+    loading: PropTypes.bool,
     history: PropTypes.array,
-    visibleItemsCount: PropTypes.number,
+    totalItemsCount: PropTypes.number,
     fetchItemsHistoryAction: PropTypes.func,
     link: PropTypes.func,
     navigate: PropTypes.func,
   };
 
   static defaultProps = {
-    items: [],
     history: [],
-    visibleItemsCount: 0,
+    loading: false,
+    totalItemsCount: 0,
     fetchItemsHistoryAction: () => {},
     link: () => {},
     navigate: () => {},
   };
 
-  getItems = () => {
-    const { items, visibleItemsCount } = this.props;
-    return items.slice(0, visibleItemsCount);
+  getHeaderItemsCount = () => {
+    const { history } = this.props;
+    let headerItemsCount = 0;
+    history.forEach((item) => {
+      if (item.resources.length > headerItemsCount) {
+        headerItemsCount = item.resources.length;
+      }
+    });
+    this.rowItemsCount = headerItemsCount;
```

Comment:
It's not clear that `getHeaderItemsCount` sets `rowItemsCount` field. Could we avoid this? Could we avoid `rowItemsCount` at all?

Reply:
Moved this logic to `state` and `getDerivedStateFromProps`